modules/apps/portal-relationship/portal-relationship-impl/src/main/java/com/liferay/portal/relationship/internal/RelationshipManagerImpl.java

```diff
@@ -141,49 +105,27 @@ public class RelationshipManagerImpl implements RelationshipManager {
 		getOutboundRelatedModels(
 			Class<T> modelClass, long primKey, Degree degree) {
 
-		List<Relationship<T>> relationships = _getRelationships(modelClass);
+		List<ClassedModel> outBoundRelatedClassedModels = new ArrayList<>();
 
-		Stream<Relationship<T>> stream = relationships.stream();
+		for (Relationship<T> relationship : _getRelationships(modelClass)) {
+			for (ClassedModel outboundRelatedClassedModel :
+					relationship.getOutboundRelatedModels(primKey)) {
 
-		List<ClassedModel> outBoundRelatedClassedModels = new ArrayList<>();
+				outBoundRelatedClassedModels.add(outboundRelatedClassedModel);
 
-		stream.forEach(
-			relationship -> {
-				List<? extends ClassedModel> outboundRelatedModels =
-					relationship.getOutboundRelatedModels(primKey);
-
-				Stream<? extends ClassedModel> outboundRelatedModelsStream =
-					outboundRelatedModels.stream();
-
-				outboundRelatedModelsStream.map(
-					outboundRelatedModel -> (ClassedModel)outboundRelatedModel
-				).forEach(
-					outboundRelatedClassedModel -> {
-						outBoundRelatedClassedModels.add(
-							outboundRelatedClassedModel);
-
-						Degree minusOneDegree = Degree.minusOne(degree);
-
-						if (minusOneDegree.getDegree() <= 0) {
-							return;
-						}
-
-						Collection<? extends ClassedModel>
-							minusOneDegreeOutboundRelatedClassedModels =
-								getInboundRelatedModels(
-									(Class)
-										outboundRelatedClassedModel.
-											getModelClass(),
-									(long)
-										outboundRelatedClassedModel.
-											getPrimaryKeyObj(),
-									minusOneDegree);
-
-						outBoundRelatedClassedModels.addAll(
-							minusOneDegreeOutboundRelatedClassedModels);
-					}
-				);
-			});
+				Degree minusOneDegree = Degree.minusOne(degree);
+
+				if (minusOneDegree.getDegree() <= 0) {
+					continue;
+				}
+
+				outBoundRelatedClassedModels.addAll(
+					getInboundRelatedModels(
```

Comment:
@W2jx The method is `getOutboundRelatedModels` while it's adding results from `getInboundRelatedModels`... Could you confirm with the author or product team whether this is a typo?